src/develop/develop.c

```diff
@@ -685,6 +685,8 @@ void dt_dev_load_image(dt_develop_t *dev, const uint32_t imgid)
   dev->first_load = 1;
   dev->image_status = dev->preview_status = dev->preview2_status = DT_DEV_PIXELPIPE_DIRTY;
 
+  dt_pthread_mutex_unlock(&darktable.db_insert);
```

Comment:
??? Double unlocking in this function? Will certainly leave the underlying mutex in a problematic state.